bin/node--test.js

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env node
+'use strict'
+
+const Module = require('node:module')
+const path = require('node:path')
+const { pathToFileURL } = require('node:url')
+const minimist = require('minimist')
+
+const { argv } = require('#internal/options')
+
+Object.assign(argv, minimist(process.argv.slice(2), {
+  boolean: ['test', 'test-only']
+}))
+if (!argv.test && !argv['test-only']) {
+  // Assume `--test` was meant if no flag was passed.
+  // TODO: emit a deprecation warning?
+  argv.test = true
+}
+process.argv.splice(1, Infinity, ...argv._)
+if (argv.test) {
+  require('#internal/main/test_runner')
+} else {
```

Comment:
why does `--test-only` require a different entry point from `--test`?

Reply:
That’s how Node.js does it – more precisely, Node.js uses a different script to startup iif `--test` was passed as a flag, independently if `--test-only` was passed or not.